integration/tests/access/rest_state_stream_test.go

```diff
@@ -0,0 +1,242 @@
+		testnet.WithAdditionalFlag("--execution-data-retry-delay=1s"),
+	)
+
+	// add the ghost (access) node config
+	ghostNode := testnet.NewNodeConfig(
+		flow.RoleAccess,
+		testnet.WithLogLevel(zerolog.FatalLevel),
+		testnet.AsGhost())
+
+	consensusConfigs := []func(config *testnet.NodeConfig){
+		testnet.WithAdditionalFlag("--cruise-ctl-fallback-proposal-duration=100ms"),
+		testnet.WithAdditionalFlag(fmt.Sprintf("--required-verification-seal-approvals=%d", 1)),
+		testnet.WithAdditionalFlag(fmt.Sprintf("--required-construction-seal-approvals=%d", 1)),
+		testnet.WithLogLevel(zerolog.FatalLevel),
+	}
+
+	nodeConfigs := []testnet.NodeConfig{
+		testnet.NewNodeConfig(flow.RoleCollection, testnet.WithLogLevel(zerolog.FatalLevel)),
+		testnet.NewNodeConfig(flow.RoleCollection, testnet.WithLogLevel(zerolog.FatalLevel)),
+		testnet.NewNodeConfig(flow.RoleExecution, testnet.WithLogLevel(zerolog.FatalLevel)),
+		testnet.NewNodeConfig(flow.RoleExecution, testnet.WithLogLevel(zerolog.FatalLevel)),
+		testnet.NewNodeConfig(flow.RoleConsensus, consensusConfigs...),
+		testnet.NewNodeConfig(flow.RoleConsensus, consensusConfigs...),
+		testnet.NewNodeConfig(flow.RoleConsensus, consensusConfigs...),
+		testnet.NewNodeConfig(flow.RoleVerification, testnet.WithLogLevel(zerolog.FatalLevel)),
+		bridgeANConfig,
+		ghostNode,
+	}
+
+	conf := testnet.NewNetworkConfig("access_api_test", nodeConfigs)
+	s.net = testnet.PrepareFlowNetwork(s.T(), conf, flow.Localnet)
+
+	// start the network
+	s.T().Logf("starting flow network with docker containers")
+	s.ctx, s.cancel = context.WithCancel(context.Background())
+
+	s.net.Start(s.ctx)
+}
+
+// TestRestEventStreaming tests event streaming route on REST
+func (s *RestStateStreamSuite) TestRestEventStreaming() {
+	ctx, cancel := context.WithTimeout(s.ctx, 1*time.Second)
+	defer cancel()
+
+	restAddr := s.net.ContainerByName(testnet.PrimaryAN).Addr(testnet.RESTPort)
+
+	s.T().Run("subscribe events", func(t *testing.T) {
+		startBlockId := flow.ZeroID
+		startHeight := uint64(0)
+		url := getSubscribeEventsRequest(restAddr, startBlockId, startHeight, nil, nil, nil)
+
+		client, err := getWSClient(ctx, url)
+		require.NoError(t, err)
+		var receivedEventsResponse []*state_stream.EventsResponse
+		eventChan := make(chan *state_stream.EventsResponse)
+
+		// Start the timeout goroutine
+		timeoutChan := make(chan struct{})
+		go func() {
+			time.Sleep(10 * time.Second) // Sleep for 10 seconds
+			close(timeoutChan)           // Signal the timeout
+		}()
+
+		go func() {
+			for {
+				resp := &state_stream.EventsResponse{}
+				err := client.ReadJSON(resp)
+				if err != nil {
+					if websocket.IsUnexpectedCloseError(err, websocket.CloseGoingAway) {
+						s.T().Logf("unexpected close error: %v", err)
+					}
+					close(eventChan) // Close the event channel when the client connection is closed
+					return
+				}
+				eventChan <- resp
+			}
+		}()
+
+		// Wait for events or timeout
+		for {
```

Comment:
I'm not totally clear on what the expected behavior is here. There seems to be 2 valid cases:
1. Test runs for 10 seconds and then all events from each response received in that time is compared against a different API call made to the same AN
2. The test received at least one event. response and the server shutdown unexpectedly.

I don't think both could be valid results. Let's pick one. Either, check that the server returns at least one response with events then end the test. Or, expect it to run for the full 10 time period.